### PR TITLE
gemma.cpp 0.1.1 (new formula)

### PR DESCRIPTION
**Pull request by xinpingwang**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/g/gemma.cpp.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  depends_on "cmake" => :build
+  depends_on "sentencepiece"
+
+  def install
+    system "cmake", "-B", "build", *std_cmake_args
+    cd "build" do
+      system "make", "gemma", "libgemma"
+      bin.install "gemma"
+      lib.install "libgemma.a"
+    end
```
Comment (SMillerDev) [inline Formula/g/gemma.cpp.rb:17]: Please use `cmake build` and `cmake install` here

File: Formula/g/gemma.cpp.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  test do
+    system "#{bin}/gemma"
```
Comment (SMillerDev) [inline Formula/g/gemma.cpp.rb:21]: Can we check some output here?

File: Formula/g/gemma.cpp.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class GemmaCpp < Formula
+  desc "Lightweight, standalone C++ inference engine for Google's Gemma models"
+  homepage "https://github.com/google/gemma.cpp"
+  url "https://github.com/google/gemma.cpp/archive/refs/tags/v0.1.0.tar.gz"
```
Comment (fxcoudert) [inline Formula/g/gemma.cpp.rb:4]: v0.1.1 is the latest

Comment (xinpingwang) [reply Formula/g/gemma.cpp.rb]: updated

File: Formula/g/gemma.cpp.rb (hunk 1 above)
Comment (xinpingwang) [reply Formula/g/gemma.cpp.rb]: tried with latest version, `cmake --build` works but `cmake --install` still have some problem.

File: Formula/g/gemma.cpp.rb [hunk 4]
```diff
@@ -1,23 +1,21 @@
...
   depends_on "sentencepiece"
 
   def install
-    system "cmake", "-B", "build", *std_cmake_args
-    cd "build" do
-      system "make", "gemma", "libgemma"
-      bin.install "gemma"
-      lib.install "libgemma.a"
-    end
+    system "cmake", "--preset", "make", *std_cmake_args
```
Comment (SMillerDev) [inline Formula/g/gemma.cpp.rb:12]: Maybe like this?
```suggestion
    system "cmake", "--preset", "make", *std_cmake_args, "-DCMAKE_INSTALL_RPATH:STRING=#{rpath}"
```

Comment (xinpingwang) [reply Formula/g/gemma.cpp.rb]: the ci still failed, maybe this not caused by rpath setting? 
image
after installed `sentencepiece`, there should be a `libsentencepiece.0.dylib` file under `/usr/local/lib` dir. And `sentencepiece` has been declared as deps.
Untitled

Comment (xinpingwang) [reply Formula/g/gemma.cpp.rb]: I got the failed reason. CI runners installed `homebrew` under `/opt`. let me have another try.

**Pull request closed without merging**